cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -515,3 +569,26 @@ func removeTombstones(fn string) ([]*zoekt.Repository, error) {
 	}
 	return tombstones, nil
 }
+
+// deleteOrTombstone deletes the provided shards in indexDir that are associated with
+// the given repoID.
+//
+// If one of the provided shards is a compound shard and the repository is contained within it,
+// the repository is tombstoned instead.
+func deleteOrTombstone(indexDir string, repoID uint32, shardMerging bool, shards ...shard) {
+	var simple []shard
```

Comment:
I see you got rid of the filtering hack, I guess just because now that it is a func it is safer that way?

Reply:
By filtering hack I assume you mean the `shardMap := getShards(indexDir)`  line? Yes, I decided to pass the list of shards associated with the ID directly since  `cleanup()` already has this information. If we're sharing the function implementation, gathering the entire `shardMap` seemed redundant.